src/components/common/UserDropdown/index.tsx

```diff
@@ -139,9 +138,8 @@ const getNetworkName = (data: Connected): string => {
 }
 
 const getWalletName = (data: Connected): string => {
-  const injectedName = getInjectedProviderName()
-  const provider = getProviderInfo(data.provider)
-  return injectedName || provider.name
+  const isMetaMask = data.provider._web3Provider && data.provider._web3Provider.isMetaMask
+  return isMetaMask ? 'MetaMask' : 'WalletConnect'
```

Comment:
If we add a provider in the future this will break. I think `provider.name` isn't configured correctly or something like that.

Reply:
I saw that _chrome dev tools_ detects the type of the object properly
![Screenshot_20200918_161545](https://user-images.githubusercontent.com/1144028/93636614-9e229d00-f9ca-11ea-9ef3-e2adc65e5e9c.png)

How to check type guards ? see https://www.typescriptlang.org/docs/handbook/advanced-types.html for more information.

If we don't get a solution for this, we will need to apply the null conditional operator (aka elvis operator ?) to check the deep properties of the object